src/models/Chemistry.ts

```diff
@@ -352,6 +397,19 @@ function checkNodesEqual(test: ASTNode, target: ASTNode, response: CheckerRespon
             newResponse.isEqual = newResponse.isEqual && test.hydrate === target.hydrate;
         } // else the 'isEqual' will already be false from the checkNodesEqual above
 
+        if (newResponse.termAtomCount) {
+            for (const [key, value] of Object.entries(newResponse.termAtomCount)) {
+                if (newResponse.atomCount) {
+                    newResponse.atomCount[key as ChemicalSymbol] = (newResponse.atomCount[key as ChemicalSymbol] ?? 0) + (value ?? 0) * test.coeff.numerator;
+                }
+                else {
+                    newResponse.atomCount = {} as Record<ChemicalSymbol, number | undefined>;
+                    newResponse.atomCount[key as ChemicalSymbol] = (value ?? 0) * test.coeff.numerator;
+                }
+                newResponse.termAtomCount = {} as Record<ChemicalSymbol, number | undefined>;
```

Comment:
Same concern for resetting the object, who's entries you are iterating over, within that iteration